Dockerfile

```diff
@@ -1,33 +1,13 @@
-FROM node:alpine3.15 as react
-
-WORKDIR /react
-
-COPY ./react/package* ./
-
-# For fix noticy about update npm. If you want to fix that, delete '#' in start next line.
-#RUN npm install -g npm@8.15.0
-
-RUN npm install
-
-COPY ./react .
-RUN npm run build
-
 FROM node:alpine3.15 as angular
-
 WORKDIR /angular
-
 COPY ./angular/package* ./
-
-RUN npm install
-
 COPY ./angular .
 RUN npm run build
 
 FROM nginx:alpine

```

Comment:
nginx version
```
FROM nginx:alpine
```

What if nginx introduces a breaking change in next release and your nginx.conf doesn't fit that change? You will have a downtime. So always use a version tagged image.